### PR TITLE
Mod Compat PR

### PR DESCRIPTION
**Pull request by f-raZ0R**
I am completely burnt out right now so there's still some TODO files for what i was going to do later, also there is no guidebook entries.
Let this go to main?

Can't be bothered to write out everything in this PR, also i changed kindling AI a bit while working on the vivisection recipe

yes there is a merge conflict but right now i don't care

### CODE REVIEW

File: src/main/java/de/dafuqs/spectrum/items/food/SedativesItem.java [hunk 1]
```diff
@@ -30,7 +32,10 @@ public ItemStack finishUsing(ItemStack stack, World world, LivingEntity user) {
 				}
 				
 			}
-			
+			if(FabricLoader.getInstance().isModLoaded("neepmeat"))
```
Comment (DaFuqs) [inline src/main/java/de/dafuqs/spectrum/items/food/SedativesItem.java]: this one should check if the integration pack is active ("neepmeat installed and integration pack not disabled")

File: src/main/resources/data/ae2/tags/blocks/blacklisted/spatial.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "replace": false,
   "values": [
-    "#spectrum:budding_blocks",
```
Comment (DaFuqs) [inline src/main/resources/data/ae2/tags/blocks/blacklisted/spatial.json:4]: This entry here mainly is:
- to give players a reason to go back to the DD from time to time
- make them not complain when they notice that moved budding blocks can not be used to generate ink, once the ink update it out (which will still take a while and will prob. be 1.21+, so eh)

File: src/main/resources/data/immersive_weathering/revelations/trees_cmy.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    "immersive_weathering:spectrum/magenta_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/light_blue_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/yellow_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/lime_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/pink_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/cyan_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/purple_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/blue_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/green_bark": "Oak Bark?",
+    "immersive_weathering:spectrum/red_bark": "Oak Bark?"
```
Comment (DaFuqs) [inline src/main/resources/data/immersive_weathering/revelations/trees_cmy.json:41]: The other spectrum blocks do use scrambled names instead.
Why should we handle it differently here?

File: src/main/resources/data/spectrum/recipes/mod_integration/affinity/TODO.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+TODO:Actually play with the mod to get a good understanding of what compat to add
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/mod_integration/affinity/TODO.txt]: The TODO files should probably not be part of the PR :P

File: src/main/resources/data/spectrum/recipes/mod_integration/create/enchantment_upgrade/capacity.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
+{
+  "type": "spectrum:enchantment_upgrade",
+  "group": "mod_compat",
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/mod_integration/create/enchantment_upgrade/capacity.json]: The field `group` is used to group and localize the unlock popups.
Since mod compat recipes are from various mods and can be about pretty much anything, they should not get grouped (else a potential bedrock knife and an enchanted book might be grouped in a single unlock popup)

File: src/main/resources/data/spectrum/recipes/mod_integration/gofish/enchanter/book_deepfry.json [hunk 6]
```diff
@@ -42,6 +42,7 @@
       ]
     }
   },
+  "required_advancement": "spectrum:unlocks/enchantments/vanilla_water",
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/mod_integration/gofish/enchanter/book_deepfry.json:45]: good catch! <3

File: src/main/resources/data/spectrum/recipes/mod_integration/neepmeat/enlightening/elder_guardian_memory.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
+  "type": "neepmeat:enlightening",
+  "item_input": {
+    "resource": "spectrum:guardian_head",
+    "amount": 1
+  },
+  "data": 1000,
+  "output": {
+    "resource": "spectrum:elder_guardian_head",
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/mod_integration/neepmeat/enlightening/elder_guardian_memory.json:9]: this is such a quirky recipe. I love it

File: src/main/resources/data/spectrum/recipes/mod_integration/neepmeat/enlightening/endermite_memory.json [hunk 8]
```diff
@@ -0,0 +1,20 @@
+{
+  "type": "neepmeat:enlightening",
+  "item_input": {
+    "resource": "spectrum:silverfish_head",
+    "amount": 1
+  },
+  "data": 1000,
+  "output": {
+    "resource": "spectrum:endermite_head",
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/mod_integration/neepmeat/enlightening/endermite_memory.json:9]: deep lore just dropped

File: src/main/resources/data/spectrum/recipes/spirit_instiller/head_fusion/fox_arctic.json [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  "ingredient2": {
+    "item": "spectrum:frostbite_essence",
+    "count": 4
+  },
+  "center_ingredient": {
+    "item": "spectrum:vegetal",
+    "count": 4
+  },
+  "result": {
+    "item": "spectrum:fox_arctic_head"
```
Comment (DaFuqs) [inline src/main/resources/data/spectrum/recipes/spirit_instiller/head_fusion/fox_arctic.json:18]: <3

**Review by DaFuqs — CHANGES_REQUESTED** [review]
Lots and lots of good stuff!
Left you a few notes.

File: src/main/resources/data/ae2/tags/blocks/blacklisted/spatial.json (hunk 2 above)
Comment (f-raZ0R) [reply src/main/resources/data/ae2/tags/blocks/blacklisted/spatial.json:4]: doesn't the resonance enchantment have these same issues anyways?

File: src/main/resources/data/immersive_weathering/revelations/trees_cmy.json (hunk 3 above)
Comment (f-raZ0R) [reply src/main/resources/data/immersive_weathering/revelations/trees_cmy.json:41]: Unlike most spectrum blocks, this is something that a player is VERY likely to encounter in natural gameplay, as stripping a colored tree will yield its respective bark regardless of progression. Additionally, my planned cloak overhaul PR (that is waiting on changes to Revelationary) will be adding more instances like this.

File: src/main/resources/data/spectrum/recipes/mod_integration/affinity/TODO.txt (hunk 4 above)
Comment (f-raZ0R) [reply src/main/resources/data/spectrum/recipes/mod_integration/affinity/TODO.txt]: Fair, I suppose :P
